src/rust/bitbox02/src/ui/ui.rs

```diff
@@ -329,12 +329,26 @@ pub fn trinary_choice_create<'a>(
     }
 
     let chosen_cb_param = Box::into_raw(Box::new(chosen_callback)) as *mut c_void;
+
+    let label_left = label_left.map(|label| crate::util::str_to_cstr_vec(label).unwrap());
+    let label_middle = label_middle.map(|label| crate::util::str_to_cstr_vec(label).unwrap());
+    let label_right = label_right.map(|label| crate::util::str_to_cstr_vec(label).unwrap());
+
     let component = unsafe {
         bitbox02_sys::trinary_choice_create(
             crate::util::str_to_cstr_vec(message).unwrap().as_ptr(), // copied in C
-            crate::util::str_to_cstr_vec(label_left).unwrap().as_ptr(), // copied in C
-            crate::util::str_to_cstr_vec(label_middle).unwrap().as_ptr(), // copied in C
-            crate::util::str_to_cstr_vec(label_right).unwrap().as_ptr(), // copied in C
+            // copied in C
+            label_left
+                .as_ref()
+                .map_or_else(|| core::ptr::null(), |label| label.as_ptr()),
```

Comment:
I don't think you need a closure here, you can just provide the function `core::ptr::null`.